src/array/reducer/move-control.spec.ts

```diff
@@ -0,0 +1,81 @@
+import { MoveArrayControlAction } from '../../actions';
+import { createFormArrayState } from '../../state';
+import { moveControlReducer } from './move-control';
+import { FORM_CONTROL_0_ID, FORM_CONTROL_ID, INITIAL_STATE_NESTED_GROUP } from './test-util';
+
+describe(`form array ${moveControlReducer.name}`, () => {
+  const testArrayValue = [ 0, 1, 2, 3, 4, 5 ];
+  const testArrayState = createFormArrayState(FORM_CONTROL_ID, testArrayValue);
+
+  it('should move controls forward', () => {
+    let action = new MoveArrayControlAction(FORM_CONTROL_ID, 2, 5);
+    let resultState = moveControlReducer(testArrayState, action);
+    expect(resultState).not.toBe(testArrayState);
+    expect(resultState.controls).not.toBe(testArrayState.controls);
+    expect(resultState.value).toEqual([ 0, 1, 3, 4, 5, 2 ]);
+
+    action = new MoveArrayControlAction(FORM_CONTROL_ID, 0, 3);
+    resultState = moveControlReducer(testArrayState, action);
+    expect(resultState.value).toEqual([ 1, 2, 3, 0, 4, 5 ]);
+  });
+
+  it('should move controls backwards', () => {
+    let action = new MoveArrayControlAction(FORM_CONTROL_ID, 5, 2);
+    let resultState = moveControlReducer(testArrayState, action);
+    expect(resultState).not.toBe(testArrayState);
+    expect(resultState.controls).not.toBe(testArrayState.controls);
+    expect(resultState.value).toEqual([ 0, 1, 5, 2, 3, 4 ]);
+
+    action = new MoveArrayControlAction(FORM_CONTROL_ID, 3, 0);
+    resultState = moveControlReducer(testArrayState, action);
+    expect(resultState.value).toEqual([ 3, 0, 1, 2, 4, 5 ]);
+  });
+
+  it('should throw an error for negative or too large indices', () => {
```

Comment:
The test name mentions negative indices, but the test doesn't contain code to check for negative indices. If it did it would reveal a bug since the implementation does not check for negative indices.